File: backend/src/main/java/org/springframework/boot/backend/entity/input/Place.java
package org.springframework.boot.backend.entity.input;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "place")
@Data
public class Place {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String country;
    private String regionCode;
    private Double latitude;
    private Double longitude;
    @ManyToOne
    @JoinColumn(name = "precision_id")
    private AccuracyOfCoordinates precision;
    private String notes;
}

File: frontend/src/app/interfaces/attributes/place.ts
import { AccuracyOfCoordinates } from "./accuracy-of-coordinates";

export interface Place {
    id: number;
    name: string;
    country: string;
    regionCode: string;
    latitude: number;
    longitude: number;
    precision: AccuracyOfCoordinates
    notes?: string;
}
